lib/ansible/modules/cloud/vmware/vmware_guest_cross_vc_clone.py

```diff
@@ -0,0 +1,330 @@
+    destination_vcenter: '{{ destination_vcenter_hostname }}'
+    destination_vcenter_username: '{{ destination_vcenter_username }}'
+    destination_vcenter_password: '{{ destination_vcenter_password }}'
+    destination_host: '{{ destination_esxi }}'
+    destination_datastore: '{{ destination_datastore }}'
+    destination_vm_folder: '{{ destination_vm_folder }}'
+    power_on: yes
+  check_mode: yes
+'''
+
+RETURN = r'''
+vm_info:
+    description: metadata about the virtual machine
+    returned: always
+    type: dict
+    sample: {
+        "vm_name": "",
+        "vcenter": "",
+        "host": "",
+        "datastore": "",
+        "vm_folder": "",
+        "power_on": ""
+    }
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.vmware import (PyVmomi, find_hostsystem_by_name,
+                                         find_datastore_by_name,
+                                         find_folder_by_name, find_vm_by_name,
+                                         connect_to_api, vmware_argument_spec,
+                                         gather_vm_facts,
+                                         wait_for_task, TaskError)
+from ansible.module_utils._text import to_native
+try:
+    from pyVmomi import vim
+except ImportError:
+    pass
+
+
+class CrossVCCloneManager(PyVmomi):
+    def __init__(self, module):
+        super(CrossVCCloneManager, self).__init__(module)
+        self.config_spec = vim.vm.ConfigSpec()
+        self.clone_spec = vim.vm.CloneSpec()
+        self.relocate_spec = vim.vm.RelocateSpec()
+        self.service_locator = vim.ServiceLocator()
+        self.destination_vcenter = self.params['destination_vcenter']
+        self.destination_vcenter_username = self.params['destination_vcenter_username']
+        self.destination_vcenter_password = self.params['destination_vcenter_password']
+
+    def get_new_vm_info(self, vm):
+        # to check if vm has been cloned in the destination vc
+        # query for the vm in destination vc
+        # get the host and datastore info
+        # get the power status of the newly cloned vm
+        info = {}
+        vm_obj = find_vm_by_name(content=self.destination_content, vm_name=vm)
+        if vm_obj is None:
+            self.module.fail_json(msg="Newly cloned VM is not found in the destination VCenter")
+        else:
+            vm_facts = gather_vm_facts(self.destination_content, vm_obj)
+            info['vm_name'] = vm
+            info['vcenter'] = self.destination_vcenter
+            info['host'] = vm_facts['hw_esxi_host']
+            info['datastore'] = vm_facts['hw_datastores']
+            info['vm_folder'] = vm_facts['hw_folder']
+            info['power_on'] = vm_facts['hw_power_status']
+        return info
+
+    def clone(self):
+        # clone the vm/template on destination VC
+        vm_folder = find_folder_by_name(content=self.destination_content, folder_name=self.params['destination_vm_folder'])
```

Comment:
```suggestion
        vm_folder = self.find_folder_by_name(content=self.destination_content, folder_name=self.params['destination_vm_folder'])
```

Reply:
I'm importing `find_folder_by_name`, so `self` will not be needed.